operators/src/adapters/raster_time.rs

```diff
@@ -239,6 +336,160 @@ where
     }
 }
 
+impl<T, F, const N: usize> Stream for RasterArrayTimeAdapter<T, F, N>
+where
+    T: Pixel,
+    F: Queryable<T>,
+    F::Stream: Stream<Item = Result<RasterTile2D<T>>> + Unpin,
+    F::Output: Future<Output = Result<F::Stream>>,
+{
+    type Item = Result<[RasterTile2D<T>; N]>;
+
+    #[allow(clippy::too_many_lines)] // TODO: refactor
+    fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        // The adapter aligns multiple two input time series by querying both sources simultaneously.
+        // All spatial tiles for the current time step are aligned, s.t. they are both valid
+        // for the same time (minimum of both). Then both queries are reset to start at the end
+        // of the previous time step.
+
+        let RasterArrayTimeAdapterProjection {
+            sources,
+            mut state,
+            query_rect,
+            num_spatial_tiles,
+        } = self.project();
+
+        loop {
+            match state.as_mut().project() {
+                ArrayStateProjection::Initial => {
+                    let array_of_futures = if let Ok(arr) = sources
+                        .iter()
+                        .map(|source| source.query(*query_rect))
+                        .collect::<Vec<_>>()
+                        .try_into()
+                    {
+                        arr
+                    } else {
+                        unreachable!("RasterArrayTimeAdapter: sources.len() != N");
+                    };
+
+                    let query_futures = ArrayJoin::new(array_of_futures);
```

Comment:
why do you not use https://docs.rs/futures/0.2.0/futures/future/fn.join_all.html ?

Reply:
`join_all` works for `Vec<X>`, I implemented something for arrays.

Reply:
can you collect them into a vector? shouldn't be any performance penalty and save quite a bit of code

Reply:
[6f7bd38](https://github.com/geo-engine/geoengine/pull/559/commits/6f7bd38ae5d0e322a3e4591df4cf62231e0ef7da)